### ISSUE TITLE
Rename attribute `shell` to `backend`

### ISSUE THREAD

**Issue by xmnlab** [osl-incubator/makim#110]
makim can works with shell interpreters e.g. bash, interpreters e.g.  python or php, and also tools/libraries e.g. nox.

so it would be much more precise if we change from shell to backend

**Comment by abhijeetSaroha** [comment]
I don't understand where I should rename `shell` to `backend`. Could you clarify exactly where the `shell` attribute needs to be renamed to `backend`?

**Comment by abhijeetSaroha** [comment]
Do I have to change here only or some other places also ?

https://github.com/osl-incubator/makim/blob/main/src/makim/core.py#L278C9-L278C71

**Comment by xmnlab** [comment]
Yep .. I think  you need to change just in this place you mentioned.
and update all the documentation, and .makim.yaml test files

**Comment by xmnlab** [comment]
done! thanks @abhijeetSaroha
